### PR TITLE
enable deployer to create/use pvcs

### PR DESCRIPTION
**Pull request by sosiouxme**
Creates PVCs if they don't exist, attaches to ES DCs.

Can go in separately from UUID work if desired

### CODE REVIEW

File: deployment/README.md [hunk 1]
```diff
@@ -209,14 +211,19 @@ as directed below.
...
-The deployer initially creates an ephemeral deployment in which all
+By default, the deployer creates an ephemeral deployment in which all
 of a pod's data will be lost any time it is restarted. For production
 use you should specify a persistent storage volume for each deployment
-of ElasticSearch. Use the `oc volume` command for adding volumes to
-deployments.
+of ElasticSearch. The deployer parameters with `PVC` in the name should
+be used for this. You can either use an existing set of PVCs (by naming
+them all in a series starting with a prefix and appending numbers starting
+at 1), or have them created requesting a specified size. This is the
```
Comment (ewolinetz) [inline deployment/README.md]: This makes sense to me from looking at the template, but can we clarify with an example? At first I wasn't sure if we should provide the deployer with `prefix-1`, `prefix`, or `prefix1`.

Create PVCs logging-es-1, logging-es-2, logging-es-3 ; provide the deployer 'logging-es-' ; etc...

File: deployment/scripts/install.sh [hunk 2]
```diff
@@ -289,19 +299,50 @@ oc delete dc,rc,pod --selector logging-infra=kibana
...
+for pvc in $(oc get persistentvolumeclaim --template='{{range .items}}{{.metadata.name}} {{end}}' 2>/dev/null); do
+  pvcs["$pvc"]=1  # note, map all that exist, not just ones labeled as supporting
+done
+for ((n=1;n<=${es_cluster_size};n++)); do
+        pvc="${ES_PVC_PREFIX}$n"
+        if [ "${pvcs[$pvc]}" != 1 -a "${ES_PVC_SIZE}" != "" ]; then # doesn't exist, create it
+          oc process logging-pvc-template -v "NAME=$pvc,SIZE=${ES_PVC_SIZE}" | oc create -f -
+          pvcs["$pvc"]=1
+        fi
+        if [ "${pvcs[$pvc]}" = 1 ]; then # exists (now), attach it
```
Comment (ewolinetz) [inline deployment/scripts/install.sh:312]: should we also check that it was bound or just let it continue on and expand our docs to help users troubleshoot PVs?

Comment (sosiouxme) [inline deployment/scripts/install.sh:312]: No, assume that everything is OK for now or will resolve itself eventually (PVC unbound is fairly discoverable). Once we get around to adding validation, status of PVCs should be checked after deployment (metrics validation PR includes this already).

File: deployment/README.md (hunk 1 above)
Comment (sosiouxme) [inline deployment/README.md]: The complexity is that they can choose a different prefix if they want for some reason. But yeah, an example from the default would be good.

**Comment by sosiouxme** [comment]
@ewolinetz README update made

**Comment by ewolinetz** [comment]
LGTM

**Comment by sosiouxme** [comment]
will create PR directly

**Pull request closed without merging**